onnx/test/inference_function_test.py

```diff
@@ -0,0 +1,64 @@
+import unittest
+from typing import Iterable, Optional, List
+
+import onnx
+import onnx.shape_inference
+
+
+def call_infer_types(
```

Comment:
I think it might be useful to promote an API like this as a wrapper around the raw C++ method (that is, make if part of the public API instead of something in just this test-file). But, for that, I don't think we would want `node` to be optional. And am not sure about `num_outputs` either.

Reply:
Sure, that definitely makes sense. We could always require a `NodeProto`, `num_outputs` was a temporary idea.